File: swig/Makefile.pl
PROJECT=IB
CCC=g++
SHARED=-shared -s

########## Target Selection  ##############################
TYP=perl
PREFIX=Pl
OPTIONS= -const  # -opt -interface $(MODULE)
EXT=.pm
############################################################

############################################################
#### This is the configuration stuff  ######################

SWIG=swig 
INCLUDE=/usr/local/include/perl5
OPT= -DSWIG -O5 -march=pentiumpro 

PIC=-fpic

############ end configuration #############################
############################################################

BIN_DIR=../$(TYP)
MODULE=$(PREFIX)$(PROJECT)
VERSION=`cat .version`
CCCFLAGS=-c $(PIC) $(OPT)
INC=-I/usr/local/include/$(TYP) -I../src -I$(INCLUDE)
LIBS=-lBSn -libDoc -libSearch -libIO -libProcess -lIBctype


target: dll 

OBJS=ib_$(TYP)wrap.o

.version: ../bin/Iindex
	./vergen
	touch $(MODULE).$(VERSION)$(EXT)
	rm $(PROJECT).$(VERSION)$(EXT)
	cp .version ../lib/$(TYP)

ib_$(TYP)wrap.cxx: ib.i my_typedefs.i .version
	@rm -f $(PROJECT)$(EXT)
	@rm -f $(PROJECT).$(VERSION)$(EXT)
	$(SWIG) -$(TYP) -c++ $(OPTIONS) -o ib_$(TYP)wrap.cxx ib.i
	rm -f $(PROJECT).$(VERSION)$(EXT)
	mv $(PROJECT)$(EXT) $(PROJECT).$(VERSION)$(EXT)
	ln -s $(PROJECT).$(VERSION)$(EXT) $(PROJECT)$(EXT)

$(MODULE): ib.i my_typedefs.i
	@rm -f $(PROJECT)$(EXT)
	@rm -f $(PROJECT).$(VERSION)$(EXT)
	$(SWIG) -c++ -$(TYP)  $(OPTIONS) -o ib_$(TYP)wrap.cxx ib.i
	rm -f $(PROJECT).$(VERSION)$(EXT)
	mv $(PROJECT)$(EXT) $(PROJECT).$(VERSION)$(EXT)
	ln -s $(PROJECT).$(VERSION)$(EXT) $(PROJECT)$(EXT)


ib_$(TYP)wrap.o: ib_$(TYP)wrap.cxx
	$(CCC) $(CCCFLAGS) $(INC) ib_$(TYP)wrap.cxx


dll: ../lib/$(TYP)/$(MODULE).so
 
../lib/$(TYP)/$(MODULE).so: $(MODULE).so
	cp -f $(MODULE).so.$(VERSION) ../lib/$(TYP)
	cp -f $(PROJECT).$(VERSION)$(EXT) ../lib/$(TYP)
	cp -f .version ../lib/$(TYP)
	rm -f ../lib/$(TYP)/$(MODULE).so
	cd ../lib/$(TYP); ln -s $(MODULE).so.$(VERSION) $(MODULE).so; touch $(PROJECT)$(EXT); rm $(PROJECT)$(EXT); ln -s $(PROJECT).$(VERSION)$(EXT) $(PROJECT)$(EXT)

$(MODULE).so: $(OBJS)
	$(CCC) $(SHARED) -o $(MODULE).so.$(VERSION) $(OBJS)  -L../lib $(LIBS) 
	rm -f $(MODULE).so
	rm -f $(PROJECT)c.so
	ln -s $(MODULE).so.$(VERSION) $(MODULE).so
	ln -s $(MODULE).so $(PROJECT)c.so

File: swig/Makefile.js
PROJECT=IB
CCC=g++
GCC_LIBDIR=/usr/local/lib/gcc-4.2.1

############################################################
#### This is the configuration stuff  ######################

SWIG=swig
TYP=java
PREFIX=J
OPTIONS= 
EXT=.java
MY_INC=-I/usr/local/jdk1.5.0/include -I/usr/local/jdk1.5.0/include/freebsd


############ end configuration #############################
############################################################
BIN_DIR=../$(TYP)
MODULE=$(PREFIX)$(PROJECT)
VERSION=`cat .version`
#PyIB
OPT=-c -fpic 
SHARED=-shared -s
INC=-I/usr/local/include/$(TYP) $(MY_INC) -I../src
LIBS=  -L$(GCC_LIBDIR) -lBSn -libDoc$(LIBX) -libSearch$(LIBX) -libIO -libProcess -lIBlocal


target: $(MODULE)c.so

OBJS=ib_$(TYP)wrap.o

.version: ../bin/Iindex
	./vergen
	touch $(MODULE).$(VERSION)$(EXT)
	rm $(MODULE).$(VERSION)$(EXT)

ib_$(TYP)wrap.cxx: ib.i my_typedefs.i .version
	@rm -f $(MODULE)$(EXT)
	$(SWIG) -c++  -$(TYP) $(OPTIONS)  -o ib_$(TYP)wrap.cxx ib.i

$(MODULE): ib.i my_typedefs.i
	@rm -f $(MODULE)$(EXT)
	$(SWIG) -c++ -$(TYP)  $(OPTIONS) -o ib_$(TYP)wrap.cxx ib.i

ib_$(TYP)wrap.o: ib_$(TYP)wrap.cxx
	$(CCC) $(OPT) $(INC) ib_$(TYP)wrap.cxx

$(MODULE)c.so: $(OBJS)
	$(CCC) $(SHARED) -o $(MODULE)c.$(VERSION).so $(OBJS) -R../lib:/opt/BSN/lib -L../lib $(LIBS) 
	rm -f $(MODULE)c.so
	ln -s $(MODULE)c.$(VERSION).so $(MODULE)c.so
	mv $(MODULE)$(EXT) $(MODULE).$(VERSION)$(EXT)
	ln -s $(MODULE).$(VERSION)$(EXT) $(MODULE)$(EXT)

